docs/source/api.rst

```diff
@@ -1,26 +1,67 @@
+~~~~~~~~~~~~~~~~~~~
 cuML API Reference
-====================
+~~~~~~~~~~~~~~~~~~~
+
+
+
+Preprocessing
+==============
+
+Model Selection and Data Splitting
+----------------------------------
+
+ .. automodule:: cuml.preprocessing.model_selection
+    :members:
+
+Label Encoding
+--------------
+
+ .. autoclass:: cuml.preprocessing.LabelEncoder
+    :members:
+
+Regression and Classification
+=============================
 
 Linear Regression
-------------------
+-----------------
 
 .. autoclass:: cuml.LinearRegression
     :members:
 
 
 Ridge Regression
------------------
+----------------
 
 .. autoclass:: cuml.Ridge
     :members:
 
+Lasso Regression
+----------------
+
+.. autoclass:: cuml.Lasso
+    :members:
+
+ElasticNet Regression
+---------------------
+
+.. autoclass:: cuml.ElasticNet
+    :members:
+    
 
 Stochastic Gradient Descent
-----------------------------
+---------------------------
 
 .. autoclass:: cuml.SGD
     :members:
 
+Random Forest
+-------------
+.. autoclass:: cuml.ensemble.RandomForestClassifier
+    :members:
+
+Clustering
+==========
+
 
 Nearest Neighbors
```

Comment:
I was surprised to see this in clustering. We don't have the classification/regression variants for it yet, but sklearn has the unsupervised version in a "neighbors" package in their docs.

Reply:
Should be under classification I guess?